homeassistant/components/bryant_evolution/climate.py

```diff
@@ -0,0 +1,246 @@
+"""Support for Bryant Evolution HVAC systems."""
+
+from datetime import timedelta
+import logging
+from typing import Any
+
+from evolutionhttp import BryantEvolutionLocalClient
+
+from homeassistant.components.climate import (
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACAction,
+    HVACMode,
+)
+from homeassistant.const import UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import BryantEvolutionConfigEntry
+from .const import CONF_SYSTEM_ZONE, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+SCAN_INTERVAL = timedelta(seconds=60)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: BryantEvolutionConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up a config entry."""
+    entities = []
+    for sz in config_entry.data[CONF_SYSTEM_ZONE]:
+        system = sz[0]
+        zone = sz[1]
+        client = config_entry.runtime_data.get(tuple(sz))
+        climate = BryantEvolutionClimate(
+            client,
+            system,
+            zone,
+            config_entry.entry_id,
+        )
+        entities.append(climate)
+    async_add_entities(entities, update_before_add=True)
+
+
+class BryantEvolutionClimate(ClimateEntity):
+    """ClimateEntity for Bryant Evolution HVAC systems.
+
+    Design note: this class updates using polling. However, polling
+    is very slow (~1500 ms / parameter). To improve the user
+    experience on updates, we also locally update this instance and
+    call async_write_ha_state as well.
+    """
+
+    _attr_has_entity_name = True
+    _attr_temperature_unit = UnitOfTemperature.FAHRENHEIT
+    _attr_supported_features = (
+        ClimateEntityFeature.TARGET_TEMPERATURE
+        | ClimateEntityFeature.TARGET_TEMPERATURE_RANGE
+        | ClimateEntityFeature.FAN_MODE
+        | ClimateEntityFeature.TURN_ON
+        | ClimateEntityFeature.TURN_OFF
+    )
+    _attr_hvac_modes = [
+        HVACMode.HEAT,
+        HVACMode.COOL,
+        HVACMode.HEAT_COOL,
+        HVACMode.OFF,
+    ]
+    _attr_fan_modes = ["auto", "low", "med", "high"]
+    _enable_turn_on_off_backwards_compatibility = False
+
+    def __init__(
+        self,
+        client: BryantEvolutionLocalClient,
+        system_id: int,
+        zone_id: int,
+        sam_id: str,
+    ) -> None:
+        """Initialize an entity from parts."""
+        self._client = client
+        self._attr_name = f"System {system_id}, Zone {zone_id}"
```

Comment:
Okay so let's dive a bit deeper onto naming. What is a system? If I understand correctly we connect with 1 SAM, which then exposes multiple systems. Which then exposes multiple zones. I think this is something nice we can model with devices and using via_device. (Not sure if you have a hub based integration yourself, but if you look into the hub device, you will most likely see a "Connected devices" card with all the devices connected. So I might think about having 1 parent device (created manually, as there is no entity for them) having 1-2 child devices (the systems) having n child devices (the zones). Then you can assign every zone to a room and work with that. Does this sound logical or would you like more info?

Reply:
lgtm, will do. This is now a hub-type integration, yeah?

Reply:
done

Reply:
shall I change the integration type to hub?